### PR TITLE
Make GHE routes testable

### LINKED ISSUE #443: Make GitHub Enterprise easily testable
Allow optional passing of `gheVersion` to `npm test`

### PR DESCRIPTION
**Pull request by d11n**
fixes #443 

##### GitHub Enterprise 2.14
```
$ GHE_VERSION=2.14 npm test -- --no-coverage

> @octokit/routes@0.0.0-development test /home/derek/code/octokit/routes
> tap --coverage --timeout 60 'test/**/*-test.js' "--no-coverage"

 PASS  test/integration/landing-page-test.js 1 OK 724ms
 PASS  test/unit/options-urls-test.js 9 OK 882ms
 PASS  test/integration/all-endpoints-test.js 964 OK 10s
 PASS  test/integration/scope-endpoints-test.js 964 OK 10s
 PASS  test/integration/single-endpoint-test.js 964 OK 10s

  🌈 SUMMARY RESULTS 🌈  

Suites:   5 passed, 5 of 5 completed
Asserts:  2902 passed, of 2902
Time:     11s
```

##### GitHub Enterprise 2.15
```
$ GHE_VERSION=2.15 npm test -- --no-coverage

> @octokit/routes@0.0.0-development test /home/derek/code/octokit/routes
> tap --coverage --timeout 60 'test/**/*-test.js' "--no-coverage"

 PASS  test/integration/landing-page-test.js 1 OK 764ms
 PASS  test/unit/options-urls-test.js 9 OK 856ms
 PASS  test/integration/all-endpoints-test.js 966 OK 11s
 PASS  test/integration/scope-endpoints-test.js 966 OK 11s
 PASS  test/integration/single-endpoint-test.js 966 OK 11s

  🌈 SUMMARY RESULTS 🌈  

Suites:   5 passed, 5 of 5 completed
Asserts:  2908 passed, of 2908
Time:     11s
```

##### GitHub Enterprise 2.16
```
$ GHE_VERSION=2.16 npm test -- --no-coverage

> @octokit/routes@0.0.0-development test /home/derek/code/octokit/routes
> tap --coverage --timeout 60 'test/**/*-test.js' "--no-coverage"

 PASS  test/integration/landing-page-test.js 1 OK 768ms
 PASS  test/unit/options-urls-test.js 9 OK 907ms
 PASS  test/integration/all-endpoints-test.js 970 OK 10s
 PASS  test/integration/scope-endpoints-test.js 970 OK 11s
 PASS  test/integration/single-endpoint-test.js 970 OK 11s

  🌈 SUMMARY RESULTS 🌈  

Suites:   5 passed, 5 of 5 completed
Asserts:  2920 passed, of 2920
Time:     11s
```

##### GitHub Enterprise 2.17
```
$ GHE_VERSION=2.17 npm test -- --no-coverage

> @octokit/routes@0.0.0-development test /home/derek/code/octokit/routes
> tap --coverage --timeout 60 'test/**/*-test.js' "--no-coverage"

 PASS  test/integration/landing-page-test.js 1 OK 794ms
 PASS  test/unit/options-urls-test.js 9 OK 858ms
 PASS  test/integration/scope-endpoints-test.js 992 OK 11s
 PASS  test/integration/all-endpoints-test.js 992 OK 11s
 PASS  test/integration/single-endpoint-test.js 992 OK 11s
     
  🌈 SUMMARY RESULTS 🌈  

Suites:   5 passed, 5 of 5 completed
Asserts:  2986 passed, of 2986
Time:     11s
```

##### GitHub Cloud
```
$ npm test -- --no-coverage

> @octokit/routes@0.0.0-development test /home/derek/code/octokit/routes
> tap --coverage --timeout 60 'test/**/*-test.js' "--no-coverage"

 PASS  test/integration/landing-page-test.js 1 OK 737ms
 PASS  test/unit/options-urls-test.js 9 OK 853ms
 PASS  test/integration/all-endpoints-test.js 1006 OK 11s
 PASS  test/integration/scope-endpoints-test.js 1006 OK 11s
 PASS  test/integration/single-endpoint-test.js 1006 OK 11s

  🌈 SUMMARY RESULTS 🌈  

Suites:   5 passed, 5 of 5 completed
Asserts:  3028 passed, of 3028
Time:     12s
```

### CODE REVIEW

**Comment by gr2m** [comment]
That looks good 👍 

Shall we add the testing to CI? The `.github/main.workflow` file would need to be updated

https://github.com/octokit/routes/blob/f3d5d7c88788514dc1b4f1bcb05f3065536118d5/.github/main.workflow#L1-L24

I wonder if you see a way to test all GHE versions without configuring the number? That way we would test al enterprise versions with a single action and wouldn’t need to update the workflow file each time an enterprise version gets added/removed

**Comment by d11n** [comment]
after going down several paths, i've landed on a pretty light touch approach in the form of adding a `test:ci` npm script (to `.github/main.workflow` as well): 97aee804d5f133c4871d2f1e0492071f338a7542

**Comment by gr2m** [comment]
Can you run a test for a single `*.JSON` for either api.github.com or one of the GHE versions now? If so, could you amend [CONTRIBUTING.md](https://github.com/octokit/routes/blob/master/CONTRIBUTING.md) please?

**Comment by d11n** [comment]
done

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -26,6 +26,12 @@ Please follow these steps to report such a problem
    TEST_URL=https://developer.github.com/v3/repos/#get ./node_modules/.bin/tap test/integration/single-endpoint-test.js
    ```
 
+   and for GitHub Enterprise:
+
+   ```
+   GHE_VERSION=2.17 TEST_URL=https://developer.github.com/enterprise/2.17/v3/repos/#get ./node_modules/.bin/tap test/integration/single-endpoint-test.js
+   ```
```
Comment (gr2m) [inline CONTRIBUTING.md]: Ideally we make this work by just passing `TEST_URL` as it also includes the GHE version, but that’s not urgent, let’s get back to it another day 👍

Comment (d11n) [reply CONTRIBUTING.md]: will track this via #446

**Pull request merged**